packages/blade/src/components/Drawer/_decisions/decisions.md

```diff
@@ -0,0 +1,147 @@
+# Drawer
+
+A drawer is a panel that slides in mostly from right side of the screen over the existing content in the viewport. It helps in providing additional details or context and can also be used to promote product features or new products.
+
+<img width="500px" src="./2024-02-07-20-04-07.png" alt="Drawer Figma Skeleton">
+
+## Design
+
+- [Drawer - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=78667%3A66663&mode=dev)
+
+## Proposed API
+
+```jsx
+<Drawer
+  isOpen={}
+  onDismiss={() => {}}
+  showOverlay
+>
+  <DrawerHeader
+    title=""
+    subtitle=""
+    leading={<DrawerHeaderIcon />}
+    titleSuffix={<DrawerHeaderBadge />}
+    trailing={
+      <>
+        <DrawerHeaderLink />
+        <DrawerHeaderButton />
+      </>
+    }
+  />
+  <DrawerBody>
+    <Slot />
+  </DrawerBody>
+<Drawer>
+```
+
+<details>
+<summary><b>Full Usage Example</b></summary>
+
+```jsx
+const MyCuteDrawer = () => {
+  const [showDrawer, setShowDrawer] = React.useState(false);
+  return (
+    <Box>
+      <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+      <Drawer
+        isOpen={showDrawer}
+        onDismiss={() => {
+          setShowDrawer(false);
+        }}
+      >
+        <DrawerHeader
+          title="Announcements"
+        />
+        <DrawerBody>
+          <FTXAnnouncement />
+          <RazorpayOnePromotions />
+          <CatPictures />
+        </DrawerBody>
+      <Drawer>
+    </Box>
+  )
+}
+
+```
+
+</details>
+
+### Props
+
+```ts
+type DrawerProps = {
+  /**
+   * Controlled state of drawer open or not
+   */
+  isOpen: boolean;
+
+  /**
+   * Dismiss handler
+   */
+  onDismiss: () => void;
+
+  /**
+   * Show or hide overlay.
+   *
+   * Also decides if clicking outside on overlay closes the drawer or not
+   */
+  showOverlay?: boolean;
+};
+
+type DrawerHeaderProps = {
+  /**
+   * Title of the Drawer
+   */
+  title: string;
+
+  /**
+   * Subtitle of the Drawer
+   */
+  subtitle?: string;
+
+  /**
+   * Leading element
+   *
+   * DrawerHeaderIcon or DrawerHeaderAsset
+   */
+  leading?: ReactNode;
+
+  /**
+   * Title suffix element
+   *
+   * DrawerHeaderBadge
+   */
+  titleSuffix?: ReactNode;
+
+  /**
+   * Title trailing element
+   *
+   * DrawerHeaderLink, DrawerHeaderButton[]
+   */
+  trailing?: ReactNode;
+};
+```
+
+Other supporting wrapper components for trailing and leading space-
+
+- DrawerHeaderBadge
+- DrawerHeaderIcon
+- DrawerHeaderAsset
+- DrawerHeaderLink
+- DrawerHeaderButton
```

Comment:
We keep creating these components for multiple components... Modal, Bottomsheet, Drawer. We might need to look into changing these into just `HeaderBadge`, `HeaderIcon`, etc. This would keep adding unnecessary repetitive components.

Reply:
true. `HeaderBadge` is a good idea. Lets discuss this with team once in next standup?

Reply:
How are these components different from normal `Badge`/`Icon`? For example, I see `_CardHeaderBadge` is same as `Badge`. Why can't we use the`Badge` component directly? 

https://github.com/razorpay/blade/blob/ba5ba9ba719d94448b0c7c9faae0c35f83d5551a/packages/blade/src/components/Card/CardHeader.tsx#L48

Reply:
These subcomponents usually have defined defaults that are different from defaults of actual component. E.g.

https://github.com/razorpay/blade/blob/ba5ba9ba719d94448b0c7c9faae0c35f83d5551a/packages/blade/src/components/ActionList/ActionListItem.tsx#L214

However in some examples like CardHeaderBadge, the defaults of wrapper are same as defaults of actual component. There it is just for consistency in API otherwise some components will be like `Badge` while some be like `CardHeaderText` etc